src/app_io.rs

```diff
@@ -305,6 +308,59 @@ impl AppIO for RealAppIO {
     fn copy_to_clipboard(&self, text: String) {
         self.egui_ctx.copy_text(text);
     }
+
+    fn start_update_checker(&self) {
+        let sender = self.msg_queue.clone();
+        let current_version = self.current_version.clone();
+        let shelv_min_version_url = format!("{}/min-version", self.shelv_api_server);
+        tokio::spawn(async move {
+            loop {
+                let client = reqwest::Client::new();
+                let response = client.get(shelv_min_version_url.clone()).send().await;
+
+                match response {
+                    Ok(response) => {
+                        let version_response = response.json::<VersionResponse>().await;
+
+                        match version_response {
+                            Ok(version_response) => {
+                                if let Some(msg) = match (
+                                    compare_versions(
+                                        &current_version,
+                                        &version_response.latest_version,
+                                    ),
+                                    compare_versions(
+                                        &current_version,
+                                        &version_response.min_version,
+                                    ),
+                                ) {
+                                    (_, Ordering::Less) => {
+                                        Some(MsgToApp::UpdateRequired(version_response.min_version))
+                                    }
+                                    (Ordering::Less, _) => Some(MsgToApp::UpdateAvailable(
+                                        version_response.latest_version,
+                                    )),
+                                    _ => None,
+                                } {
+                                    sender.send(msg).unwrap();
+                                }
+                            }
+                            Err(err) => {
+                                // Keep check though, hopefully someone notices they messed up and fixes it.
+                                println!("Error parsing min version response: {err:#?}");
+                                // TODO maybe log parsing errors to sentry?
+                            }
+                        }
+                    }
+                    Err(err) => {
+                        // Keep checking though, might be intermittent network issue.
+                        println!("Error requesting min version response: {err:#?}");
+                    }
+                }
+                tokio::time::sleep(std::time::Duration::from_secs(10)).await;
```

Comment:
did you mean to put just 10 secs? we wanted to have 10*60, right? I assume it is a left over from your local debugging